Add support for C++20 Compilation (#2785)
The-OpenROAD-Project/OpenROAD: docs/contrib/Logger.md

automatic calls to the logger which will then prefix the appropriate
severity type to the message.

## C++20 Requirements

In C++20 the logger messages are checked during compile time which introduces
restrictions around rutime format strings. See [docs](https://fmt.dev/latest/api.html#compile-time-format-string-checks)

> OpenROAD uses SPDLog which usues fmt_lib under the hood below is an example of
> what is no longer allowed.

Reviewer: ```suggestion
OpenROAD uses `spdlog` which uses `fmt_lib` under the hood. Below is an example of
what is no longer allowed.
```

> In order to make use of runtime format strings we have introduced a 

Reviewer: ```suggestion
In order to make use of runtime format strings, we have introduced a 
```

> `FMT_RUNTIME` macro in Logger.h. You should use this any time that you are 
> passing a dynamic string as the format string 

Reviewer: ```suggestion
`FMT_RUNTIME` macro in Logger.h. You should use this macro any time you
pass a dynamic string as the format string 
```
